theano/sandbox/cuda/extra_ops.py

```diff
@@ -352,7 +355,10 @@ def c_support_code_apply(self, node, nodename):
     def c_code(self, node, nodename, inames, onames, sub):
         x, = inames
         z, = onames
+
+        # We assume array has been already flattened if needed.
         axis = self.axis if self.axis is not None else 0
+
```

Comment:
Is there a way I can infer the number of dimension of from x? If not, I'll modify the string `code` to use `CudaNdarray_NDIM(%(x)s)`